### PR TITLE
Add smart ptrs

### PR DESCRIPTION
**Pull request by DanielGuramulta**
This pull request adds back smart pointers in the SDK.

### CODE REVIEW

File: apps/uvc-app/uvc-gadget.cpp [hunk 1]
```diff
@@ -284,7 +284,8 @@ void stopHandler(int code) { stop.store(true); }
  * V4L2 streaming related
  */
 
-static int v4l2_process_data(LocalDevice *device, uvc_device *udev) {
+static int v4l2_process_data(std::shared_ptr<LocalDevice> device,
```
Comment (grigorovici-andreea) [inline apps/uvc-app/uvc-gadget.cpp]: uvc_device *udev 
Astea de ce nu vor fi schimbate?

File: sdk/src/frame.cpp [hunk 2]
```diff
@@ -5,27 +5,17 @@ namespace aditof {
...
-    if (m_impl)
-        delete m_impl;
-}
+Frame::~Frame() = default;
 
-Frame::Frame(Frame &&other) noexcept {
-    this->m_impl = other.m_impl;
-    other.m_impl = nullptr;
-};
+Frame::Frame(Frame &&other) noexcept = default;
```
Comment (grigorovici-andreea) [inline sdk/src/frame.cpp]: Oare nu ar fi mai ok sa fie ca aici, fara nume (other) : System::System(System &&)

Comment (mhennerich) [reply sdk/src/frame.cpp]: Friendly reminder - Interaction on github should be in plain English please

File: apps/uvc-app/uvc-gadget.cpp (hunk 1 above)
Comment (DanielGuramulta) [reply apps/uvc-app/uvc-gadget.cpp]: The uvc gadget was initially implemented using C, the main purpose of this PR is to add smart pointers to the SDK api.

File: sdk/src/frame.cpp (hunk 2 above)
Comment (DanielGuramulta) [reply sdk/src/frame.cpp]: Modified to Frame::Frame(Frame &&). Thanks.
